tests/core/test_signal.py

```diff
@@ -119,3 +122,18 @@ async def test_set_and_wait_for_value():
     assert not st.done
     set_sim_put_proceeds(sim_signal, True)
     assert await time_taken_by(st) < 0.1
+
+
+async def test_create_soft_signal():
```

Comment:
Could: I would suggest splitting these into separate tests by either:
* Using `pytest.mark.parametrize`
* Have a helper function that is the test and call it with each case